### PR TITLE
SecurityHotspotTest: link only required references

### PR DESCRIPTION
**Pull request by costin-zaharia-sonarsource**
When running the hotspots, link only the required references instead of doing a union of all the references needed by all hotspots

### CODE REVIEW

File: analyzers/tests/SonarAnalyzer.UnitTest/Rules/Hotspots/CookieShouldBeHttpOnlyTest.cs [hunk 1]
```diff
@@ -99,5 +98,8 @@ public void CookiesShouldBeHttpOnly_WithWebConfigValueSetToFalse()
         private static IEnumerable<MetadataReference> GetAdditionalReferences_NetCore() =>
             NuGetMetadataReference.MicrosoftAspNetCoreHttpFeatures(Constants.NuGetLatestVersion);
 #endif
+
+        internal static IEnumerable<MetadataReference> AdditionalReferences =>
```
Comment (andrei-epure-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/Rules/Hotspots/CookieShouldBeHttpOnlyTest.cs:102]: nitpick internal should be above private (you can make this change in your next PR though)

File: analyzers/tests/SonarAnalyzer.UnitTest/Rules/Hotspots/ExecutingSqlQueriesTest.cs [hunk 2]
```diff
@@ -40,26 +40,26 @@ public class ExecutingSqlQueriesTest
...
 
         internal static IEnumerable<MetadataReference> GetReferencesNet46(string sqlServerCeVersion) =>
-            Enumerable.Empty<MetadataReference>()
-                .Concat(FrameworkMetadataReference.SystemData)
-                .Concat(FrameworkMetadataReference.SystemDataOracleClient)
-                .Concat(NuGetMetadataReference.SystemDataSqlServerCe(sqlServerCeVersion)
-                .Concat(NuGetMetadataReference.MySqlData("8.0.22"))
-                .Concat(NuGetMetadataReference.MicrosoftDataSqliteCore())
-                .Concat(NuGetMetadataReference.SystemDataSQLiteCore()));
+            NetStandardMetadataReference.Netstandard
```
Comment (andrei-epure-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/Rules/Hotspots/ExecutingSqlQueriesTest.cs:56]: for my education: why was it working before?

**Review by andrei-epure-sonarsource — APPROVED** [review]
LGTM!

File: analyzers/tests/SonarAnalyzer.UnitTest/Rules/Hotspots/CookieShouldBeHttpOnlyTest.cs (hunk 1 above)
Comment (andrei-epure-sonarsource) [reply analyzers/tests/SonarAnalyzer.UnitTest/Rules/Hotspots/CookieShouldBeHttpOnlyTest.cs:102]: I realize now that the private one is inside a preprocessor directive, so it doesn't apply

File: analyzers/tests/SonarAnalyzer.UnitTest/Rules/Hotspots/ExecutingSqlQueriesTest.cs (hunk 2 above)
Comment (costin-zaharia-sonarsource) [reply analyzers/tests/SonarAnalyzer.UnitTest/Rules/Hotspots/ExecutingSqlQueriesTest.cs:56]: In this test file we use `Verifier.VerifyAnalyzer` which adds the Netstandard if missing. See https://github.com/SonarSource/sonar-dotnet/blob/f75a387985c158f654dee052b644bf5b60c41b24/analyzers/tests/SonarAnalyzer.UnitTest/TestFramework/SolutionBuilder.cs#L77-L81

`Verifier.VerifyNoIssueReported` which is used by `SecurityHotspotTest` is not doing this so we have to add it manually. I think in the long term we will have to unify a bit the behavior.

**Pull request merged**